server/graphql/v2/object/CurrencyFxRate.ts

```diff
@@ -0,0 +1,36 @@
+import { GraphQLBoolean, GraphQLFloat, GraphQLNonNull, GraphQLObjectType } from 'graphql';
+import { GraphQLDateTime } from 'graphql-scalars';
+
+import { Currency } from '../enum';
+import { CurrencyFxRateSourceType } from '../enum/CurrencyFxRateSourceType';
+
+const CurrencyFxRate = new GraphQLObjectType({
+  name: 'CurrencyFxRate',
+  description: 'Fields for a currency fx rate',
+  fields: () => ({
+    percentage: {
+      type: new GraphQLNonNull(GraphQLFloat),
+      description: 'Amount of this item',
+    },
+    source: {
+      type: new GraphQLNonNull(CurrencyFxRateSourceType),
+      description: 'Where does the FX rate comes from',
+    },
+    fromCurrency: {
+      type: new GraphQLNonNull(Currency),
+    },
```

Comment:
`from` is what we use internally, I've seen some API like Fixer using `base`.

Reply:
Is it necessary though? I guess this should always be the same as `amount.currency` ?

Reply:
It's not necessary here as we indeed have the same info in `Amount`. But the idea was to be able to re-use this object and have something able to completely describe a currency conversion regardless of the context. We'll also need this `fromCurrency` if we want rates to be cached at some point (id would be something like `${fromCurrency}-${toCurrency}-${source}-${date}`.

Regarding the name itself, I personally find the `fromCurrency`/`toCurrency` pair predictable and clearer than `base`. To compare:
- Wise uses `sourceCurrency`/`targetCurrency`
- PayPal uses an object like:

```ts
  currency_conversion: {
    to_amount: { value: number; currency: string };
    from_amount: { value: number; currency: string };
    exchange_rate: string;
  };
```